### PR TITLE
Mc 29 improve tag handling

### PR DESCRIPTION
**Pull request by haub94**

### CODE REVIEW

File: Merge Conflict/Assets/Scripts/TagHandler.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Author(s):     Markus Haubold
+Date:          2024-03-04
+Version:       V1.0
+TODO:          - ongoing: add new tags
+**********************************************************************************************************************/
+
+using UnityEngine;
+
+//add all tags which are used in unity!
+public static class TagHandler
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/TagHandler.cs:14]: rename to simply "Tags" for simplicity

Comment (Xipit) [inline Merge Conflict/Assets/Scripts/TagHandler.cs:14]: refactor class to enum + extension methods 
(-> https://stackoverflow.com/a/5985710)

File: Merge Conflict/Assets/Scripts/TagHandler.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        component,
+        trashcan,
+        ConveyorBelt,
+        ConveyorBeltDiagonal
+    }
+
+    public static bool CompareTags(GameObject gameObject, PossibleTags possibleTag)
+    {
+        if (gameObject.CompareTag(possibleTag.ToString())) { return true; }
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/TagHandler.cs:26]: simplify to 
return gameobject.CompareTag()

File: Merge Conflict/Assets/Scripts/TagHandler.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public enum PossibleTags
+    {
+        component,
+        trashcan,
+        ConveyorBelt,
+        ConveyorBeltDiagonal
+    }
+
+    public static bool CompareTags(GameObject gameObject, PossibleTags possibleTag)
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/TagHandler.cs:24]: when refactrong to enum + extensions, the method name needs to reflect it. Goal is that the Method Call reads like a sentence.
e.g. "Tags.component.CheckGameObject()"

**Review by Xipit — CHANGES_REQUESTED** [review]
good and functional start, but can be simplified and made easier to work with

File: Merge Conflict/Assets/Scripts/TagHandler.cs (hunk 1 above)
Comment (haub94) [reply Merge Conflict/Assets/Scripts/TagHandler.cs:14]: According to this [guideline](https://github.com/ktaranov/naming-convention/blob/master/C%23%20Coding%20Standards%20and%20Naming%20Conventions.md#c-coding-standards-and-naming-conventions) for naming classes i will rename it with Tag.

**Review by Xipit — APPROVED** [review]

**Pull request merged**